lib/utils/remote-build.ts

```diff
@@ -210,6 +213,28 @@ async function cancelBuildIfNecessary(build: RemoteBuild): Promise<void> {
 	}
 }
 
+/**
+ * Return a callback function that takes a tar-stream Pack object as argument
+ * and uses it to add the '.balena/registry-secrets.json' metadata file that
+ * contains usernames and passwords to private docker registries. The builder
+ * will remove the file from the tar stream and use the secrets to pull base
+ * images from users' private registries.
+ * @param registrySecrets JS object containing registry usernames and passwords
+ * @returns A callback function, or undefined if registrySecrets is empty
+ */
+function getTarStreamCallbackForRegistrySecrets(
+	registrySecrets: RegistrySecrets,
+): { (pack: Pack): void } | undefined {
+	if (Object.keys(registrySecrets).length) {
```

Comment:
Can we get a test against `> 0` please? I know it's pretty much equivalent, but it makes it easier to parse quickly.

Reply:
Done